### ISSUE TITLE
Problem with 'training refurbished' mask

### ISSUE THREAD

**Issue by ed-fish** [zhoubenjia/GFSLT-VLP#27]
Hi, I have an issue when I use the noise injection with the MASK id being out of range. 

`/opt/conda/conda-bld/pytorch_1729647350383/work/aten/src/ATen/native/cuda/Indexing.cu:1308: indexSelectLargeIndex: block: [290,0,0], thread: [24,0,0] Assertion `srcIndex < srcSelectDimSize` failed`

The only way to resolve this is to remove the noise injection from the code as such:

```py
  if self.training_refurbish:
            # masked_tgt = utils.NoiseInjecting(tgt_batch, self.args.noise_rate, noise_type=self.args.noise_type, random_shuffle=self.args.random_shuffle, is_train=(self.phase=='train'), mask_token=PAD_IDX)
            masked_tgt = tgt_batch
            with self.tokenizer.as_target_tokenizer():
                masked_tgt_input = self.tokenizer(masked_tgt, return_tensors="pt", padding = True,  truncation=True)
            return src_input, tgt_input, masked_tgt_input
        return src_input, tgt_input
```
        
Is there a known solution to this issue. 

Thanks
